hub/core/dataset/dataset.py

```diff
@@ -1048,6 +1055,7 @@ def size_approx(self):
         return size
 
     @hub_reporter.record_call
+    @invalid_view_op
```

Comment:
I think this decorator should be above `@hub_reporter.record_call` for reporting to work properly.